internal/core/client/basetx/basetx.go

```diff
@@ -17,20 +19,55 @@ type Transactor struct {
 	registry *registry.ValistRegistry
 }
 
-func NewTransactor(valist *valist.Valist, registry *registry.ValistRegistry) *Transactor {
-	return &Transactor{valist, registry}
+func NewTransactor(eth bind.ContractBackend, valistAddress, registryAddress common.Address) (*Transactor, error) {
+	valist, err := contract.NewValist(valistAddress, eth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize valist contract: %v", err)
+	}
+
+	registry, err := contract.NewRegistry(registryAddress, eth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize registry contract: %v", err)
+	}
+
+	return &Transactor{
+		valist:   valist,
+		registry: registry,
+	}, nil
+}
+
+func (t *Transactor) CreateOrganizationTx(txopts *gasless.TransactOpts, metaCID string) (*types.Transaction, error) {
+	return t.valist.CreateOrganization(&txopts.TransactOpts, metaCID)
 }
 
-// TransactOpts returns default transaction options.
-func TransactOpts(account accounts.Account, wallet accounts.Wallet, chainID *big.Int) *bind.TransactOpts {
-	return &bind.TransactOpts{
-		From: account.Address,
-		Signer: func(address common.Address, tx *types.Transaction) (*types.Transaction, error) {
-			if address != account.Address {
-				return nil, bind.ErrNotAuthorized
-			}
+func (t *Transactor) SetOrganizationMetaTx(txopts *gasless.TransactOpts, orgID common.Hash, metaCID string) (*types.Transaction, error) {
+	return t.valist.SetOrgMeta(&txopts.TransactOpts, orgID, metaCID)
+}
 
-			return wallet.SignTx(account, tx, chainID)
-		},
-	}
+func (t *Transactor) LinkOrganizationNameTx(txopts *gasless.TransactOpts, orgID common.Hash, name string) (*types.Transaction, error) {
```

Comment:
Nice move reconsolidating these into this file, that was a bit confusing before 😎